documentation/proc-pages/physics-models/fusion_reactions/plasma_reactions.md

```diff
@@ -0,0 +1,236 @@
+- `f_dd_proton_branching (float)`: Branching ratio of tritium producing D-D reactions to 3He ones.
+- `sigmav_dt_average (float)`: Average fusion reaction rate $<\sigma v>$ for D-T.
+- `dhe3_power_density (float)`: Fusion power density produced by the D-3He reaction.
+- `dd_power_density (float)`: Fusion power density produced by the D-D reactions.
+- `dt_power_density (float)`: Fusion power density produced by the D-T reaction.
+- `alpha_power_density (float)`: Power density of alpha particles produced.
+- `charged_power_density (float)`: Power density of charged particles produced.
+- `neutron_power_density (float)`: Power density of neutrons produced.
+- `fusion_rate_density (float)`: Fusion reaction rate density.
+- `alpha_rate_density (float)`: Alpha particle production rate density.
+- `proton_rate_density (float)`: Proton production rate density.
+
+All variables above are initialized to be 0.0.
+
+---------------------
+
+### Deuterium Branching fraction | `deuterium_branching()`
+
+Calculates the relative rate of tritium producing D-D reactions to 3He ones based on the volume averaged ion temperature[^1].
+Valid for ion temperatures between 0.5 keV and 200 keV.
+The deviation of the fit from the R-matrix branching ratio is always smaller than 0.5%.
+
+$$
+ = 1.02934 - 8.3264\times 10^{-3}\langle T_{\text{e}} \rangle + 1.7631\times 10^{-4}\langle T_{\text{e}} \rangle^2 \\
+-1.8201\times 10^{-6}\langle T_{\text{e}} \rangle^3 + 6.9855\times 10^{-9}\langle T_{\text{e}} \rangle^4
+$$
+
+<figure markdown>
+![D-D reaction branching ratio](./deuterium_branching_plot.png){ width = "100"}
+<figcaption>Figure 1: Ratio of the neutron producing to proton producing D-D reaction branches based on the ion temperature .</figcaption>
+</figure>
+
+#### Attributes Updated
+The method updates the following attributes:
+
+- `f_dd_proton_branching (float)`: Branching ratio of tritium producing D-D reactions to 3He ones.
+
+-----------------------
+
+### Calculate fusion reactions
+
+There are 4 key functions for calculating the fusion reaction for the plasma. They are `dt_reaction()`, `dhe3_reaction()`, `dd_helion_reaction()` and `dd_triton_reaction()`. They all perform the same key calculations below but with their own specific values for their reactions.
+
+#### Detailed Steps
+1. **Initialize Bosch-Hale Constants**: Initializes the Bosch-Hale constants for the required reaction using predefined reaction constants stored in the BoschHaleConstants dataclass.
+2. **Calculate Fusion Reaction Rate**: Uses Simpson's rule to integrate the fusion reaction rate over the plasma profile.
+3. **Calculate Fusion Power Density**: Compute the fusion power density produced by the given reaction. Using the reaction energy calculated and stored in `constants.f90`. The reactant density is is given by $\mathtt{f\_deuterium, f\_tritium}$ or $\mathtt{f\_helium3}$ multiplied by the volume averaged ion density.
```

Comment:
Duplicate 'is': "reactant density is is given" ->  "reactant density is given "

Reply:
Changed